packages/expo-cli/src/commands/client/index.js

```diff
@@ -0,0 +1,101 @@
+import chalk from 'chalk';
+import CliTable from 'cli-table';
+import { Credentials, User, ApiV2 } from 'xdl';
+
+import urlOpts from '../../urlOpts';
+import * as appleApi from '../build/ios/appleApi';
+import { runAction, travelingFastlane } from '../build/ios/appleApi/fastlane';
+import * as credentials from '../build/ios/credentials';
+import promptForCredentials from '../build/ios/credentials/prompt/promptForCredentials';
+import selectDistributionCert from './selectDistributionCert';
+import selectPushKey from './selectPushKey';
+import generateBundleIdentifier from './generateBundleIdentifier';
+import log from '../../log';
+import prompt from '../../prompt';
+
+export default program => {
+  program
+    .command('client:ios')
+    .option(
+      '--apple-id <login>',
+      'Apple ID username (please also set the Apple ID password as EXPO_APPLE_PASSWORD environment variable).'
+    )
+    .asyncAction(async options => {
+      let authData = await appleApi.authenticate(options);
+      let user = await User.getCurrentUserAsync();
+      let bundleIdentifier = generateBundleIdentifier(authData.team.id);
+      let context = {
+        ...authData,
+        bundleIdentifier,
+        experienceName: 'Expo',
+        username: user ? user.username : null,
+      };
+      await appleApi.ensureAppExists(context);
+
+      let distributionCert = await selectDistributionCert(context);
+      let pushKey = await selectPushKey(context);
+
+      let email;
+      if (user) {
+        email = user.email;
+      } else {
+        ({ email } = await prompt({
+          name: 'email',
+          message: 'Please enter an email address to notify, when the build is completed:',
+          default: user ? user.email : undefined,
+          filter: value => value.trim(),
+          validate: value => (/@/.test(value) ? true : "That doesn't look like a valid email."),
```

Comment:
I think we need a better validation here.

Reply:
The only way to reliably validate an email address is to try sending email to it. Failing that, it's better to keep the validation as simple as possible to avoid false negatives. (E.g. some web forms don't allow addresses like `myemail+label@example.com` or the more obscure TLDs.) Even the full [regular expression to validate an email address according to RFC822](http://www.ex-parrot.com/~pdw/Mail-RFC822-Address.html) won't help if I have a typo in any individual component of the address.

Reply:
Well, yes, but `@` is a valid email according to this condition. We could at least check whether there is anything before and after the `@` character.